lib/std/fs.zig

```diff
@@ -1122,6 +1130,17 @@ pub const Dir = struct {
     pub fn deleteFileZ(self: Dir, sub_path_c: [*:0]const u8) DeleteFileError!void {
         os.unlinkatZ(self.fd, sub_path_c, 0) catch |err| switch (err) {
             error.DirNotEmpty => unreachable, // not passing AT_REMOVEDIR
+            error.AccessDenied => |e| switch (builtin.os.tag) {
+                // non-Linux POSIX systems return EPERM when trying to delete a directory, so
+                // we need to handle that case specifically and translate the error
+                .macosx, .ios, .freebsd, .netbsd, .dragonfly => {
```

Comment:
I guess we could also assume the same behaviour as in macOS in watchOS and tvOS?

Reply:
Probably true, but unless I'm reading this code wrong, `fs.zig` will fail to compile for those OS's, so I just went with the set of OS's that are currently used/referenced in the fs implementation:

https://github.com/ziglang/zig/blob/374e3e42e0de10d21406c077599cfc4a6a813497/lib/std/fs.zig#L43-L53

Basically just kicking the can down the road, but oh well.

Reply:
SGTM!